src/posix/platform/hdlc_interface.cpp

```diff
@@ -123,10 +123,10 @@ namespace ot {
 namespace PosixApp {
 
 HdlcInterface::HdlcInterface(Callbacks &aCallbacks)
-    : mCallbacks(aCallbacks)
+    : SpinelInterface()
+    , mCallbacks(aCallbacks)
     , mSockFd(-1)
-    , mRxFrameBuffer()
-    , mHdlcDecoder(mRxFrameBuffer, HandleHdlcFrame, this)
+    , mHdlcDecoder(GetRxFrameBuffer(), HandleHdlcFrame, this)
```

Comment:
Can we move mRxFrameBuffer to protected?

Reply:
Good idea. I have updated the code.